lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,34 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument if it is a
+    // magic literal expression.
+    if (parameterBindings[paramIdx].empty()) {
+      if (!callee || !isa<FuncDecl>(callee))
+        continue;
```

Comment:
Note that there are other kinds of decls that can have default arguments, for example initialisers and subscripts. I believe you should be able to use `swift::getParameterAt` here, which will handle those cases for you.

Reply:
The full list is at least, functions, initializers, subscripts, and enum elements. It would be nice to avoid duplicating that here

Reply:
Thanks, I've done that now!